### PR TITLE
DAOS-7274 test: Added config_generate and updated storage_scan and ne…

### PR DESCRIPTION
**Pull request by shimizukko**
…twork_scan

config_generate() will be used in the upcoming test. The test is implemented,
but not included in this PR to make it shorter.

Updated storage_scan() and network_scan() to use --json. Updated affected tests.
They're updated because the upcoming config generate test will use these commands
with --json.

Tag - Test
pool_create_tests - pool/create_capacity_test.py, pool/create_test.py
server_reformat - control/dmg_system_reformat.py
server_restart - server/daos_server_restart.py
network_scan - control/dmg_network_scan.py
dmg_storage_scan_scm - control/dmg_storage_scan_scm.py
ssd_socket - control/ssd_socket.py

Quick-build: true
Skip-unit-tests: true
Test-tag-hw-large: pr pool_create_tests server_reformat server_restart
Test-tag-hw-small: pr network_scan dmg_storage_scan_scm ssd_socket
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/control/ssd_socket.py [hunk 1]
```diff
@@ -66,19 +66,24 @@ def test_scan_ssd(self):
         :avocado: tags=all,small,full_regression,hw,control,ssd_socket
         """
         # Call dmg storage scan --verbose and get the PCI addresses.
-        data = self.get_dmg_command().storage_scan(verbose=True)
-        pci_addrs = list(data[self.hostlist_servers[0]]["nvme"].keys())
-        self.log.info("Testing PCI addresses: %s", pci_addrs)
+        storage_out = self.get_dmg_command().storage_scan(verbose=True)
+
```
Comment (shimizukko) [inline src/tests/ftest/control/ssd_socket.py:70]: if storage_out["status"] != 0:
    self.fail("dmg storage scan failed!")

File: src/tests/ftest/control/ssd_socket.py [hunk 2]
```diff
@@ -66,19 +66,24 @@ def test_scan_ssd(self):
...
+        # Get nvme_devices and scm_namespaces list that are buried. There's a
+        # uint64 hash of the strcut under HostStorage.
+        temp_dict = storage_out["response"]["HostStorage"]
+        struct_hash = list(temp_dict.keys())[0]
```
Comment (shimizukko) [inline src/tests/ftest/control/ssd_socket.py:74]: Check if there's only one element, or one hash.

**Review by phender — APPROVED** [review]

**Review by dinghwah — DISMISSED** [review]
LGTM

**Review by dinghwah — APPROVED** [review]
LGTM

**Review by phender — APPROVED** [review]

**Comment by sylviachanoiyee** [comment]
I'm out this week and landing is still limited to TB1 and stability. Please ping any gatekeepers when landing opens.

**Comment by shimizukko** [comment]
The PR tests and the tests that are affected by my change passed.

**Pull request merged**